src/components/POLine/Item/ContributorForm.js

```diff
@@ -64,14 +70,18 @@ class ContributorForm extends Component {
             fullWidth
             label={<FormattedMessage id="ui-orders.itemDetails.contributor" />}
             name={`${elem}.contributor`}
+            onChange={(e, value) => this.props.onChangeField(value, `${elem}.contributor`)}
           />
         </Col>
         <Col xs={2} style={{ paddingTop: '4px' }}>
           <br />
           <Button
             buttonStyle="danger"
             data-test-remove-contributor-button
-            onClick={() => this.removeFields(fields, index)}
+            onClick={() => {
+              this.removeFields(fields, index);
+              this.props.setInstanceId();
```

Comment:
`setInstanceId` looks like wrong name to me. you call it `set...`, but provide no arguments...

Reply:
fixed